src/OpenTelemetry.Exporter.Console/ConsoleExporter.cs

```diff
@@ -17,42 +17,28 @@
 using System;
 using System.Diagnostics;
 using System.Linq;
-using System.Text.Json;
-using System.Text.Json.Serialization;
+#if NET461 || NETSTANDARD2_0
+using OpenTelemetry.Logs;
+#endif
 using OpenTelemetry.Resources;
 using OpenTelemetry.Trace;
 
 namespace OpenTelemetry.Exporter
 {
-    public class ConsoleExporter : BaseExporter<Activity>
+    public class ConsoleExporter<T> : BaseExporter<T>
```

Comment:
So now I can do ConsoleExporter<some_custom_type> and it will emit nothing? I don't see a lot of value in having the generic, TBH. Why don't we just have `ConsoleLogExporter : BaseExporter<LogRecord>` & `ConsoleActivityExporter : BaseExporter<Activity>`? There's really not that much code being duplicated.

Reply:
OR how about something like...
```csharp
public class ConsoleExporter<T> : BaseExporter<T>
{
   public ConsoleExporter(Action<T> writeToConsole) { ... }

   public override ExportResult Export(in Batch<T> batch)
   {
      foreach (var item in batch) {
         this.writeToConsole(item);
     }
   }
}

public class ConsoleLogExporter : ConsoleExporter<LogRecord>
{
   public ConsoleLogExporter () : base(WriteLogToConsole) { }

   private static void WriteLogToConsole(LogRecord logRecord) {
      var rightPaddingLength = 30;

      Console.WriteLine($"{"LogRecord.TraceId:".PadRight(rightPaddingLength)}{logRecord.TraceId}");
      Console.WriteLine($"{"LogRecord.SpanId:".PadRight(rightPaddingLength)}{logRecord.SpanId}");
      Console.WriteLine($"{"LogRecord.Timestamp:".PadRight(rightPaddingLength)}{logRecord.Timestamp:yyyy-MM-ddTHH:mm:ss.fffffffZ}");
      Console.WriteLine($"{"LogRecord.EventId:".PadRight(rightPaddingLength)}{logRecord.EventId}");
      Console.WriteLine($"{"LogRecord.CategoryName:".PadRight(rightPaddingLength)}{logRecord.CategoryName}");
      Console.WriteLine($"{"LogRecord.LogLevel:".PadRight(rightPaddingLength)}{logRecord.LogLevel}");
      Console.WriteLine($"{"LogRecord.TraceFlags:".PadRight(rightPaddingLength)}{logRecord.TraceFlags}");
      Console.WriteLine($"{"LogRecord.State:".PadRight(rightPaddingLength)}{logRecord.State}");
      if (logRecord.Exception is { })
      {
         Console.WriteLine($"{"LogRecord.Exception:".PadRight(rightPaddingLength)}{logRecord.Exception?.Message}");
      }

      Console.WriteLine();
   }
}
```

Reply:
I don't think you can add an Exporter of a type anything other than Activity or LogRecord. To add an exporter for traces, you have to use the TracerProviderBuilder AddProcessor(**BaseProcessor<Activity>** processor) method of TracerProviderBuilder class. Here, both SimpleExportProcessor and BatchExportProcessor will enforce you to only use Activity as the type.

Similarly, to add an exporter to Logs, you have to use the extension method OpenTelemetryLoggerOptions AddProcessor(**BaseProcessor<LogRecord>** processor). Here, both SimpleExportProcessor and BatchExportProcessor will enforce you to only use LogRecord as the type.

Reply:
That's a good point. I still feel like it is misusing generics and we could improve it though. If it was internal, no prob, but it is part of the public API so we'll have to live with it forever.

@cijothomas @reyang Any thoughts?

Reply:
No strong opinion from my side.

I agree that having a switch-case (or if-else) in generic method (e.g. if T is `Activity` do this, if T is `LogRecord` do that) is not very idiomatic / performant.